samples/bluetooth/mesh/blob_transfer/src/blob_shell.c

```diff
@@ -0,0 +1,542 @@
+		return 1;
+	}
+	shell_print(shell, "Cancel to receive blob");
+
+	return 0;
+}
+
+static int cmd_receive_progress(const struct shell *shell, size_t argc, char *argv[])
+{
+	int progress;
+
+	if (!bt_mesh_is_provisioned()) {
+		shell_print(shell, "The mesh node is not provisioned. Please provision the mesh "
+				   "node before using the blob transfer.");
+		return 1;
+	}
+
+	progress = blob_rx_progress();
+
+	if (progress < 0) {
+		shell_error(shell, "BLOB Transfer RX is NOT in active");
+		return 1;
+	}
+	shell_print(shell, "Current transfer progress: %d", progress);
+
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(
+	blob_receive_cmds,
+	SHELL_CMD_ARG(init, NULL, "Set flash area to receive blob data. <offset>", cmd_receive_init,
+		      2, 0),
+	SHELL_CMD_ARG(start, NULL,
+		      "Start to receive blob data. <blob_id> [-t ttl] [-b base_timeout]",
+		      cmd_receive_start, 2, 4),
+	SHELL_CMD(cancel, NULL, "Cancel transfer", cmd_receive_cancel),
+	SHELL_CMD(progress, NULL, "Show progress of transfer", cmd_receive_progress),
+	SHELL_SUBCMD_SET_END);
+
+static int cmd_send(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (argc == 1) {
+		shell_help(shell);
+		/* shell returns 1 when help is printed */
+		return 1;
+	}
+
+	shell_error(shell, "%s unknown parameter: %s", argv[0], argv[1]);
+
+	return 0;
+}
+
+static int cmd_receive(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (argc == 1) {
+		shell_help(shell);
+		/* shell returns 1 when help is printed */
+		return 1;
+	}
+
+	shell_error(shell, "%s unknown parameter: %s", argv[0], argv[1]);
+
+	return 0;
+}
+
+static int _cmd_blob(blob_cb cb, const struct shell *shell, size_t argc, char *argv[])
+{
+	int err = 0;
+	off_t offset;
+	size_t size;
+
+	offset = shell_strtoul(argv[1], 16, &err);
+
+	if (err) {
+		shell_error(shell, "Unable to parse input string argument");
+		return err;
+	}
+
+	size = shell_strtoul(argv[2], 16, &err);
+
+	if (err) {
+		shell_error(shell, "Unable to parse input string argument");
+		return err;
+	}
+
+	if (offset < STORAGE_START) {
+		shell_error(shell, "Offset must be %#x or greater", STORAGE_START);
+		return -1;
+	} else if (size == 0) {
+		shell_error(shell, "Size 0 are not allowed");
+		return -1;
+	} else if (offset + size > STORAGE_END) {
+		shell_error(shell, "Allowed storage range: %#x - %#x", STORAGE_START, STORAGE_END);
+		return -1;
+	} else if (offset % STORAGE_ALIGN != 0) {
+		shell_error(shell, "Offset must be aligned in %#x", STORAGE_ALIGN);
+		return -1;
+	}
+
+	cb(get_flash_area_id(), offset, size);
+
+	return 0;
+}
+
+inline static int cmd_read_blob(const struct shell *shell, size_t argc, char *argv[])
+{
+	return _cmd_blob(read_flash_area, shell, argc, argv);
+}
+
+inline static int cmd_write_blob(const struct shell *shell, size_t argc, char *argv[])
+{
+	return _cmd_blob(write_flash_area, shell, argc, argv);
+}
+
+inline static int cmd_hash_blob(const struct shell *shell, size_t argc, char *argv[])
```

Comment:
We sort of have all of this already:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/shell/blob.c#L569 

Couldn't you use that?

Reply:
I misunderstood the conversation we had on DevZone. I'll fix it by using the implementation you mentioned.